wallet/src/components/IbcTransfer.tsx

```diff
@@ -0,0 +1,497 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { useEffect, useMemo, useState } from 'react';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import InputAdornment from '@mui/material/InputAdornment';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import { parseAsValue } from '@agoric/ui-components';
+import { withApplicationContext } from '../contexts/Application';
+import { ibcAssets } from '../util/ibc-assets';
+import { stringifyPurseValue } from '@agoric/ui-components';
+import ArrowDownward from '@mui/icons-material/ArrowDownward';
+import { Box } from '@mui/system';
+import { fromBech32 } from '@cosmjs/encoding';
+import { queryBankBalances } from '../util/queryBankBalances';
+import { assertIsDeliverTxSuccess } from '@cosmjs/stargate';
+import { CircularProgress, Link, Snackbar, Typography } from '@mui/material';
+import Petname from './Petname';
+import type { PurseInfo } from '../service/Offers';
+import type { KeplrUtils } from '../contexts/Provider';
+import type { Petname as PetnameType } from '@agoric/smart-wallet/src/types';
+import { sendIbcTokens, withdrawIbcTokens } from '../util/ibcTransfer';
+
+export enum IbcDirection {
+  Deposit,
+  Withdrawal,
+}
+
+const unmodifiableAddressStyle = {
+  width: 420,
+  '& .Mui-disabled': {
+    color: 'rgba(0,0,0,0.6)',
+  },
+  '& input.Mui-disabled': {
+    color: 'rgba(0,0,0,0.86)',
+  },
+};
+
+const titlePreposition = {
+  [IbcDirection.Deposit]: 'from',
+  [IbcDirection.Withdrawal]: 'to',
+};
+
+const agoricAddressLabel = {
+  [IbcDirection.Deposit]: 'To',
+  [IbcDirection.Withdrawal]: 'From',
+};
+
+const remoteChainAddressLabel = {
+  [IbcDirection.Deposit]: 'From',
+  [IbcDirection.Withdrawal]: 'To',
+};
+
+interface Params {
+  purse?: PurseInfo;
+  direction: IbcDirection;
+  handleClose: () => void;
+  keplrConnection: KeplrUtils;
+}
+
+const agoricExplorerPath = 'agoric';
+
+const useRemoteChainAccount = (brandPetname?: PetnameType) => {
+  const ibcAsset =
+    typeof brandPetname === 'string' ? ibcAssets[brandPetname] : undefined;
+
+  const [remoteChainAddress, setRemoteChainAddress] = useState('');
+  const [remoteChainSigner, setRemoteChainSigner] = useState(null);
+  const [remoteChainBalance, setRemoteChainBalance] = useState<bigint | null>(
+    null,
+  );
+
+  const connectWithKeplr = async () => {
+    assert(ibcAsset);
+    // @ts-expect-error window keys
+    const { keplr } = window;
+    const offlineSigner = await keplr.getOfflineSignerOnlyAmino(
+      ibcAsset.chainInfo.chainId,
+    );
+    const accounts = await offlineSigner.getAccounts();
+    setRemoteChainAddress(accounts[0].address);
```

Comment:
why is it safe to pick the first? does this code assume the array has only one element? if so best to assert.

Reply:
Yea keplr only returns one address. I think an assert might be too harsh but added a warning so we can be aware if some behavior changes, it shouldn't break anything though.